source/library/CabalGild/Unstable/Type/Condition.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module CabalGild.Unstable.Type.Condition where
+
+import qualified CabalGild.Unstable.Extra.CharParsing as Parse
+import qualified Distribution.Compat.CharParsing as Parse
+import qualified Distribution.Parsec as Parsec
+import qualified Text.Parsec as P
+import qualified Text.Parsec.Expr as PE
+import qualified Text.PrettyPrint as PrettyPrint
+
+data Condition a
+  = Par (Condition a)
+  | Not (Condition a)
+  | And (Condition a) (Condition a)
+  | Or (Condition a) (Condition a)
+  | Lit Bool
+  | Var a
+  deriving (Eq, Show)
+
+parseCondition :: Parsec.ParsecParser a -> Parsec.ParsecParser (Condition a)
+parseCondition parseVariable = Parsec.PP $ \csv -> do
+  let operators :: (P.Stream s m Char) => PE.OperatorTable s u m (Condition b)
+      operators =
+        [ [PE.Prefix (Not <$ Parse.char '!' <* Parse.spaces)],
+          [PE.Infix (And <$ Parse.string "&&" <* Parse.spaces) PE.AssocRight],
+          [PE.Infix (Or <$ Parse.string "||" <* Parse.spaces) PE.AssocRight]
+        ]
+  Parse.spaces
+  PE.buildExpressionParser operators $
+    Parsec.unPP
+      ( Parse.choice
+          [ Par <$> Parse.parens (parseCondition parseVariable),
+            Not <$> (Parse.token "!" *> parseCondition parseVariable),
+            Lit <$> Parse.try parseLit,
```

Comment:
This needed `try` because trying to parse `flag(x)` would fail because the `f` would be taken for `false`, but then it would fail and not back track.